ui/src/shared/utils/debouncer.ts

```diff
@@ -0,0 +1,43 @@
+type F = (...args: any[]) => any
+
+export interface Debouncer {
+  call: (f: F, ms: number) => void
+  cancel: (f: F) => void
+  cancelAll: () => void
+}
+
+class DefaultDebouncer implements Debouncer {
+  private timers
+
+  constructor() {
+    this.timers = {}
+  }
+
+  public call(f, ms) {
```

Comment:
any need to type these params?

Reply:
should be handled by the `implements` keyword